Tests/AutoCollection/HttpRequestParser.tests.ts

```diff
@@ -202,7 +202,7 @@ describe("AutoCollection/HttpRequestParser", () => {
             headers: {
                 host: "bing.com",
                 "x-forwarded-for": "123.123.123.123",
-                "cookie": "ai_user=cookieUser|time;ai_session=cookieSession|time;ai_authUser=cookieAuthUser|time",
+                "cookie": `ai_user=cookieUser|time;ai_session=cookieSession|time;ai_authUser=${encodeURI("cookieAuthUser|time")}`,
```

Comment:
Can you add an actual test for encoded vs decoded?, what happen to the scenario where value is not a URI?

Reply:
Sure - I had assumed based on Nev's comments that we'd always receive an encoded URI here. The good news is that the `encodeURI()` method if passed a standard string will just return that same string. To throw an exception here someone would have to pass in a string containing a "%" without two hex digits following it. In that case I believe it's safe to assume that we've received a malformed URI and should catch that exception.